models/User.php

```diff
@@ -352,6 +353,23 @@ public function attemptConfirmation($code)
     }
 
     /**
+     * Generates a new password and sends it to the user.
+     *
+     * @param string $code Confirmation code.
+     *
+     * @return boolean
+     */
+    public function resendPassword()
+    {
+        $generated_password = Password::generate(8);
```

Comment:
Moreover there is no need in `generated_password` property at all. Just assign value of Password::generate() directly to $this->password